assets/scripts/utils/image.js

```diff
@@ -0,0 +1,79 @@
+import { queryClosestParent } from './html';
+
+const LAZY_LOADED_IMAGES = []
+
+export function loadImage(url, options = {}) {
+    return new Promise((resolve, reject) => {
+        const $img = new Image();
+        if (options.crossOrigin) $img.crossOrigin = options.crossOrigin;
+
+        const loadCallback = () => {
+            resolve({
+                element: $img,
+                ...getImageMetadata($img),
+            });
+        }
+
+        if($img.decode) {
+            $img.src = url
+            $img.decode().then(loadCallback).catch(e => {
+                reject(e)
+            })
+        } else {
+            $img.onload = loadCallback
+            $img.onerror = (e) => {
+                reject(e);
+            };
+            $img.src = url
+        }
+    });
+}
+
+export function getImageMetadata($img) {
+    return {
+        url: $img.src,
+        width: $img.naturalWidth,
+        height: $img.naturalHeight,
+        ratio: $img.naturalWidth / $img.naturalHeight,
+    };
+}
+
+/**
+ * Lazy load images
+ *
+ * @param {node} | $el
+ * @param {string} | url
+ * @param {function} | callback
+ */
+export async function lazyLoadImage($el, url, callback = () => {}) {
```

Comment:
1. We should strive for better block comments.
2. Should this be the `default` export?
3. Is it better to test if `callback` is a function rather than generate noop functions?
4. Should we promisify `requestAnimationFrame()` instead of a callback API?

```suggestion
/**
 * Lazy load the given image.
 *
 * @param {HTMLImageElement} $el      - The image element.
 * @param {?string}          url      - The URI to lazy load into $el.
 *     If falsey, the value of the `data-src` attribute on $el will be used as the URI.
 * @param {?function}        callback - A function to call when the image is loaded.
 */
export async function lazyLoadImage($el, url, callback) {
```

Reply:
Fixed all this changes except #4 "promisify `requestAnimationFrame()`". I'll check with @arnvvd